packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js

```diff
@@ -43,6 +44,7 @@ class CloudFormation {
       this.pollQueue = new BottleNeck({ minTime: 100, maxConcurrent: CFN_MAX_CONCURRENT_REQUEST });
       this.pollQueueStacks = [];
       this.stackEvents = [];
+      this.readStackEventsCalls = 0;
```

Comment:
what is lifespan of `CloudFormation` object ?
if it's shared then you might want to refactor readStackEvents(stackName) to use private readStackEvents(stackNAme, timeout) and recurse similarly to how pollStack works.